### ISSUE TITLE
Make a more uniform Milestone structure, or better documentation of Milestones

### ISSUE THREAD

**Issue by vthornheart-bng** [Bungie-net/api#812]
This was a winter wishlist ask - while I don't see ourselves changing the existing Milestone structure due to backcompat reasons, I could see us adding more documentation about it - or potentially deprecating the current ones and splitting them into multiple separate concepts (now that there are so many different things we consider to be "milestones")

**Comment by nine13tech** [comment]
This has been a stick in my code for the longest time, and it's where my project started back in the first few months of D1.
I would love to see some reorganization in the return data. Like group things that have a start/end date data and things that don't into two different sections. Usually, the ones that do not have start and end data are the ongoing ever present things to do like the raid.
I would also like to see uniform milestones for the more general activity - like:
Seasonal Crucible Bucket: that would include activities like Crimson Doubles, Iron Banner. but it would be one milestone that returns all the activities that can be looked up
Conversely, if there was a set milestone ID to look for. Maybe I just cannot wrap my head around the fact that IB-Clash and IB-Control are two different things instead of just being IB with an activity type of control. But I would see that kind of organization across the public milestone data.
thanks for the consideration. As always, you guys make this game even more amazing by allowing us all this access. Thank you.
9er

**Comment by vthornheart-bng** [comment]
Yeah, it's unfortunate that Milestones ended up in basically the same rut that Advisors did in D1.  I'd definitely like to restructure this now as well: it was made particularly bad (both for maintenance and in the resulting data) when the game moved away from Milestones as indicators of what you're doing now.  It's a square peg in a round hole.

**Comment by ArkahnX** [comment]
Not knowing what the future of destiny will bring, id almost say I'd rather leave milestones in the sad state that they are for the current iteration of the game, but spend more time on a modular system somewhere down the line (next major release, next change to milestones/challenges, etc)

**Comment by vthornheart-bng** [comment]
Yeah, I will definitely want to overhaul this system for the next major iteration of the API: and indeed, I don't think I'll likely have time to do so before then.  I only hope I get time for the next iteration!
